src/advanced-concepts/scalability/_mo/main3.mo

```diff
@@ -0,0 +1,33 @@
+// ANCHOR: a
+// `main.mo`
+import Principal "mo:base/Principal";
+import Buffer "mo:base/Buffer";
+import Error "mo:base/Error";
+
+import User "actor-class";
+
+actor {
+    let users = Buffer.Buffer<User.User>(1);
+
+    public func newUser(name : Text) : async Principal {
```

Comment:
If you wanted to, `newUser` could actually have return type `async User`, providing a strongly typed actor that other canisters can interact with directly.